src/fpylll/contrib/simple_bkz.py

```diff
@@ -110,3 +110,59 @@ def svp_reduction(self, kappa, block_size):
             self.m.remove_last_row()
 
         return False
+
+class DBKZReduction(BKZReduction):
+    
+    def bkz_loop(self, block_size, min_row, max_row):
+        self.m.update_gso()
+        clean = True
+        for kappa in range(max_row - block_size, min_row - 1, -1):
+            clean &= self.dsvp_reduction(kappa, block_size)
+        clean &= BKZReduction.bkz_loop(self, block_size, min_row, max_row)
+        return clean
+
+    def dsvp_reduction(self, kappa, block_size):
+        """FIXME! briefly describe function
+
+        :param kappa:
+        :param block_size:
+        :returns:
+        :rtype:
+
+        """
+        clean = True
+
+        self.lll_obj(0, kappa, kappa + block_size)
+        if self.lll_obj.nswaps > 0:
+            clean = False
+
+        max_dist, expo = self.m.get_r_exp(kappa + block_size - 1, kappa + block_size - 1)
+        max_dist = 1.0/max_dist
+        expo *= -1.0
+        delta_max_dist = self.lll_obj.delta * max_dist
+        
+        solution, max_dist = Enum.enumerate(self.m, max_dist, expo, kappa, kappa + block_size, None, dual=True)
+        if max_dist >= delta_max_dist:
+            return clean
+        
+        with self.m.row_ops(kappa, kappa+block_size):
+            def euclid(pair1, pair2):
```

Comment:
An advantage of putting this on the top level would be that we can test it better.